cleanup/azure.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+set -eu -o pipefail
+
+run_az_deletion_command() {
+  # Check the DRYRUN environment variable
+  if [ "${DRYRUN:-true}" = "false" ] || [ "${DRYRUN:-true}" = "no" ]
+  then
+    # Execute command "as it"
+    echo "command that would be executed : "
+    echo "az group" "$@"
+    az group "$@"
+  else
+    # Execute command with the "--dry-run"
+    subcommand="$1"
+    shift
+    echo "== DRY RUN (show instead of ${subcommand}):"
+    echo "az group show $@"
+    az group show "$@" 2>&1 | grep -v 'Request would have succeeded' || true
```

Comment:
🤔 I'm not sure to understand why the command here is not the same as the one on line 11 (in the "real deletion" case branch)?

In the AWS script, there is a bunch of argument manipulation because aws CLI provides a `--dry-run` flag. Alas the `az` CLI does not.

I have 2 minor annoyances with the current behavior:
- By using the `az group show`, the output show a lot of verbose information that does not feel useful: I would expect only the resource groups names (and/or IDs) to be shown in dry run.
- When using dry run (by default) it ain't easy to really determine the command that should be run in production once deletion time is there.


WDYT about, instead of

```
== DRY RUN (show instead of delete):
az group show --name pkr-Resource-Group-zzxpwiae3b
{
  "id": "/subscriptions/dff2ec18-6a8e-405c-8e45-b7df7465acf0/resourceGroups/pkr-Resource-Group-zzxpwiae3b",
  "location": "eastus",
  "managedBy": null,
  "name": "pkr-Resource-Group-zzxpwiae3b",
  "properties": {
    "provisioningState": "Succeeded"
  },
  "tags": {
    "build_type": "dev",
    "imageplatform": "amd64",
    "imagetype": "jenkins-agent-windows-2019",
    "scm_ref": "b14bece",
    "timestamp": "20220127170111",
    "version": "0.0.1"
  },
  "type": "Microsoft.Resources/resourceGroups"
}
== DRY RUN (show instead of delete):
az group show --name pkr-Resource-Group-8dntm8cpki
{
  "id": "/subscriptions/dff2ec18-6a8e-405c-8e45-b7df7465acf0/resourceGroups/pkr-Resource-Group-8dntm8cpki",
  "location": "eastus",
  "managedBy": null,
  "name": "pkr-Resource-Group-8dntm8cpki",
  "properties": {
    "provisioningState": "Succeeded"
  },
  "tags": {
    "build_type": "dev",
    "imageplatform": "amd64",
    "imagetype": "jenkins-agent-windows-2019",
    "scm_ref": "c9d27b9",
    "timestamp": "20220211184117",
    "version": "0.0.1"
  },
  "type": "Microsoft.Resources/resourceGroups"
}
```


that we would have something like:

```
== DRY RUN
= Command that would be executed without dry-run:
az group delete --name pkr-Resource-Group-zzxpwiae3b --yes
= Command that would be executed without dry-run:
az group delete --namepkr-Resource-Group-8dntm8cpki
```
?

Reply:
👍🏻 love it